### PR TITLE
Removes kubectl get dependency on Kubernetes core

### PR DESCRIPTION
**Pull request by seans3**
* Removes `kubectl get` dependency on Kubernetes core (one of three remaining dependencies)
* Moves `pkg/printers/tableprinter.go` and `pkg/printers/tabwriter.go` to `cli-runtime/pkg/printers`
* Moves `PrintOptions` to `cli-runtime/pkg/printers`
* Removes redundant `TabWriter`, and updates references to the remaining `TabWriter` in cli-runtime.

/kind cleanup
/sig cli
/area kubectl
/priority important-soon

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/cli-runtime/pkg/printers/interface.go [hunk 1]
```diff
@@ -35,3 +36,19 @@ type ResourcePrinter interface {
 	// Print receives a runtime object, formats it and prints it to a writer.
 	PrintObj(runtime.Object, io.Writer) error
 }
+
+// PrintOptions struct defines a struct for various print options
+type PrintOptions struct {
```
Comment (liggitt) [inline staging/src/k8s.io/cli-runtime/pkg/printers/interface.go]: can we remove the declaration in pkg/printers/interface.go? I don't see any references to the old type

**Review by dixudx — APPROVED** [review]
Please fix the nit that @liggitt mentioned. Others look good to me.

File: staging/src/k8s.io/cli-runtime/pkg/printers/interface.go (hunk 1 above)
Comment (seans3) [reply staging/src/k8s.io/cli-runtime/pkg/printers/interface.go]: Fixed.

**Pull request merged**